src/material/datepicker/date-range-picker.ts

```diff
@@ -23,8 +22,9 @@ import {MAT_RANGE_DATE_SELECTION_MODEL_PROVIDER, DateRange} from './date-selecti
   encapsulation: ViewEncapsulation.None,
   providers: [MAT_RANGE_DATE_SELECTION_MODEL_PROVIDER]
 })
-export class MatDateRangePicker<D>
-  extends MatDatepickerBase<MatDateRangeInput<D>, DateRange<D>, D> {
+export class MatDateRangePicker<D> extends
+  MatDatepickerBase<MatDatepickerControl<D> & {comparisonStart: D|null, comparisonEnd: D|null},
```

Comment:
Would it be good to make a common interface that `MatDateRangeInput` implements? That way it would enforce that the 2 classes are in agreement on the type of these additional properties

Reply:
Done.